core/src/main/java/io/confluent/rest/validation/JacksonMessageBodyProvider.java

```diff
@@ -40,10 +42,13 @@
 public class JacksonMessageBodyProvider extends JacksonJaxbJsonProvider {
 
   public JacksonMessageBodyProvider() {
-    setMapper(new ObjectMapper());
+	ObjectMapper mapper = new ObjectMapper();
+	mapper.enable(DeserializationFeature.FAIL_ON_READING_DUP_TREE_KEY);
+    setMapper(mapper);
   }
 
   public JacksonMessageBodyProvider(ObjectMapper mapper) {
+	mapper.enable(DeserializationFeature.FAIL_ON_READING_DUP_TREE_KEY);
```

Comment:
Forcibly overriding this seems to be counter to the goal of this constructor, which is to allow the user to get control over the exact settings of the `ObjectMapper`. Is there any reason this needs to be overridden here? If the caller chose to leave `FAIL_ON_READING_DUP_TREE_KEY` off, is there any reason not to respect that?

This might require adjustments to kafka-rest to address the issue that triggered this PR, but keep in mind this class is used across a few projects and some might not need to be as restrictive and might prefer being a bit more liberal with the input they accept.